kratos/elements/distance_calculation_flux_based_element.cpp

```diff
@@ -0,0 +1,502 @@
+
+    const GeometryType& r_geometry = this->GetGeometry();
+
+    for(unsigned int i=0; i<NumNodes; ++i){
+        const Node<3>& rNode = r_geometry[i];
+        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(DISTANCE,rNode);
+        // KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(ADJOINT_VECTOR_1,rNode);
+
+        // Check that required dofs exist
+        KRATOS_CHECK_DOF_IN_NODE(DISTANCE,rNode);
+    }
+
+    // If this is a 2D problem, check that nodes are in XY plane
+    if ( TDim == 2){
+        for (unsigned int i=0; i<NumNodes; ++i) {
+            if (std::abs(r_geometry[i].Z())>1e-9)
+                KRATOS_ERROR << "Node " << r_geometry[i].Id() << "has non-zero Z coordinate." << std::endl;
+        }
+    }
+
+    return out;
+
+    KRATOS_CATCH("");
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Public I/O
+
+template <unsigned int TDim , unsigned int TNumNodes >
+std::string DistanceCalculationFluxBasedElement<TDim, TNumNodes >::Info() const
+{
+    std::stringstream buffer;
+    buffer << "DistanceCalculationFluxBasedElement" << Dim << "D" << NumNodes << "N #" << this->Id();
+    return buffer.str();
+}
+
+template <unsigned int TDim, unsigned int TNumNodes >
+void DistanceCalculationFluxBasedElement<TDim,TNumNodes>::PrintInfo(
+    std::ostream &rOStream) const
+{
+    rOStream << this->Info() << std::endl;
+}
+
+template <unsigned int TDim , unsigned int TNumNodes >
+void DistanceCalculationFluxBasedElement< TDim, TNumNodes  >::EquationIdVector(EquationIdVectorType &rResult, const ProcessInfo &rCurrentProcessInfo) const
+{
+    const GeometryType& r_geometry = this->GetGeometry();
+
+    unsigned int LocalIndex = 0;
+
+    if (rResult.size() != LocalSize)
+        rResult.resize(LocalSize, false);
+
+    for (unsigned int i = 0; i < NumNodes; ++i){
+        rResult[LocalIndex++] = r_geometry[i].GetDof(DISTANCE).EquationId();
+    }
+}
+
+
+template <unsigned int TDim , unsigned int TNumNodes >
+void DistanceCalculationFluxBasedElement< TDim , TNumNodes >::GetDofList(DofsVectorType &rElementalDofList, const ProcessInfo &rCurrentProcessInfo) const
+{
+    const GeometryType& r_geometry = this->GetGeometry();
+
+     if (rElementalDofList.size() != LocalSize)
+         rElementalDofList.resize(LocalSize);
+
+     unsigned int LocalIndex = 0;
+
+     for (unsigned int i = 0; i < NumNodes; ++i){
+         rElementalDofList[LocalIndex++] = r_geometry[i].pGetDof(DISTANCE);
+     }
+}
+
+template <unsigned int TDim , unsigned int TNumNodes >
+double DistanceCalculationFluxBasedElement< TDim, TNumNodes >::ComputeH(BoundedMatrix<double,NumNodes, TDim>& DN_DX)
+{
+        double h=0.0;
+        for(unsigned int i=0; i<NumNodes; i++){
+            double h_inv = 0.0;
+            for(unsigned int k=0; k<TDim; k++)
+            {
+                h_inv += DN_DX(i,k)*DN_DX(i,k);
+            }
+            h = std::max(h, 1.0 / h_inv);
+        }
+        h = std::sqrt(h);
+        return h;
+}
```

Comment:
As commented above, I'd remove this in favor of the `ElementSizeCalculator`.

Aside of this, I think that the gradient-based element size calculator cannot be used for the hexas. I'd suggest using the average element size in there instead.

Reply:
yes, i've switched to that one, thanks